tm2/pkg/bft/privval/state/state.go

```diff
@@ -0,0 +1,325 @@
+package state
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/gnolang/gno/tm2/pkg/amino"
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	tmtime "github.com/gnolang/gno/tm2/pkg/bft/types/time"
+	"github.com/gnolang/gno/tm2/pkg/errors"
+	osm "github.com/gnolang/gno/tm2/pkg/os"
+)
+
+// Step is the step in the consensus process.
+type Step uint8
+
+const (
+	StepNone      Step = 0 // Undefined step
+	StepPropose   Step = 1 // Only used in proposal signing
+	StepPrevote   Step = 2 // Only used in vote signing
+	StepPrecommit Step = 3 // Only used in vote signing
+)
+
+// A vote type is either stepPrevote or stepPrecommit.
+func VoteTypeToStep(voteType types.SignedMsgType) Step {
+	switch voteType {
+	case types.PrevoteType:
+		return StepPrevote
+	case types.PrecommitType:
+		return StepPrecommit
+	default:
+		panic("Unknown vote type")
+	}
+}
+
+// FileState stores the state of the last signing operation in a file.
+// NOTE: keep in sync with gno.land/cmd/gnoland/secrets.go
+type FileState struct {
```

Comment:
Just a note, this was migrated out from `privval/file.go`?

Reply:
Done: https://github.com/gnolang/gno/pull/3554/commits/aeaec5bf7de1e96e6583360e8acce05128361622